### ISSUE TITLE
Missing ARM Images for Go Base Image

### ISSUE THREAD

**Issue by HenriBeck** [aws/aws-lambda-base-images#37]
Currently, the docker image with the ARM arch is missing for Golang while all the other base images have an ARM image.

Would it be possible to also add ARM support to Golang?

Docker Hub: https://hub.docker.com/search?q=aws-lambda&type=image&architecture=arm%2Carm64

```sh
docker pull amazon/aws-lambda-go
docker inspect amazon/aws-lambda-go
```

**Comment by elijahgloveroptus** [comment]
Use Docker Image public.ecr.aws/lambda/provided:al2-arm64
or provided:al2

**Comment by tmokmss** [comment]
This is also the case for Node.js images.

```sh
# ERROR
docker pull --platform linux/arm64 amazon/aws-lambda-nodejs:22

# OK
docker pull --platform linux/arm64 public.ecr.aws/lambda/nodejs:22
```

**Comment by leandrodamascena** [comment]
> This is also the case for Node.js images.
> 
> ```shell
> # ERROR
> docker pull --platform linux/arm64 amazon/aws-lambda-nodejs:22
> 
> # OK
> docker pull --platform linux/arm64 public.ecr.aws/lambda/nodejs:22
> ```

Hi @tmokmss! I'm just curious to understand the impact on your workload/pipeline/automation caused by this image being published to ECR and not dockerhub. Is this some requirement/regulation? Rate pull limit? Can you explain me a little bit more this?

Thanks

**Comment by tmokmss** [comment]
Hi @leandrodamascena. Not much impact for us. We just successfully switched to ECR for ARM images. I was expecting Docker Hub images to work exactly the same way as ECR ones, so it was surprising though.

Imo if it is difficult to properly maintain Docker Hub images, it might be a good idea to deprecate them to avoid confusing users.

**Comment by leandrodamascena** [comment]
Thanks for clarifying this @tmokmss! I'll try to contact the team responsible for publishing images to see if I can get more information on this.

But I'm glad you can use the public ECR for this.

**Comment by leandrodamascena** [comment]
Hey @tmokmss! Images for arm64 were published in dockerhub. Can you please take a look?

![Image](https://github.com/user-attachments/assets/3eb5304b-845a-4cf4-a92d-cd5f2f8e0f52)

**Comment by tmokmss** [comment]
@leandrodamascena Cool, thanks! I confirmed I can now use `amazon/aws-lambda-nodejs:22-arm64` image.

However, one thing to improve; I want to use the same tag (`22`) for both arm64 and amd64 platforms. This way we can just pass `--platform` arg to switch platform, which is already possible on ECR public.

So,

```sh
# OK
docker pull --platform linux/amd64 public.ecr.aws/lambda/nodejs:22
docker pull --platform linux/arm64 public.ecr.aws/lambda/nodejs:22
docker pull --platform linux/amd64 amazon/aws-lambda-nodejs:22
docker pull --platform linux/arm64 amazon/aws-lambda-nodejs:22-arm64 # not ideal

# ERROR but ideally we can do this
docker pull --platform linux/arm64 amazon/aws-lambda-nodejs:22
```

**Comment by leandrodamascena** [comment]
> 
> However, one thing to improve; I want to use the same tag (`22`) for both arm64 and amd64 platforms. This way we can just pass `--platform` arg to switch platform, which is already possible on ECR public.
> 
> So,
> 
> # OK
> docker pull --platform linux/amd64 public.ecr.aws/lambda/nodejs:22
> docker pull --platform linux/arm64 public.ecr.aws/lambda/nodejs:22
> docker pull --platform linux/amd64 amazon/aws-lambda-nodejs:22
> docker pull --platform linux/arm64 amazon/aws-lambda-nodejs:22-arm64 # not ideal
> 
> # ERROR but ideally we can do this
> docker pull --platform linux/arm64 amazon/aws-lambda-nodejs:22

Yes, that's a good report to improve customer experience. I've already reported it to the team and while I can't provide any expected timeframe for when that will happen, I'll let you know if there are any updates.
